### ISSUE TITLE
PyArray_Descr.names undocumented

### ISSUE THREAD

**Issue by pitrou** [numpy/numpy#10585]
In https://docs.scipy.org/doc/numpy/reference/c-api.types-and-structures.html#c.PyArray_Descr, the `names` field in the C `PyArray_Descr` struct is undocumented.

**Comment by pelson** [comment]
The full definition https://github.com/numpy/numpy/blob/master/numpy/core/include/numpy/ndarraytypes.h#L591-L651 is at. On master it is currently:

```
typedef struct _PyArray_Descr {
        PyObject_HEAD
        /*
         * the type object representing an
         * instance of this type -- should not
         * be two type_numbers with the same type
         * object.
         */
        PyTypeObject *typeobj;
        /* kind for this type */
        char kind;
        /* unique-character representing this type */
        char type;
        /*
         * '>' (big), '<' (little), '|'
         * (not-applicable), or '=' (native).
         */
        char byteorder;
        /* flags describing data type */
        char flags;
        /* number representing this type */
        int type_num;
        /* element size (itemsize) for this type */
        int elsize;
        /* alignment needed for this type */
        int alignment;
        /*
         * Non-NULL if this type is
         * is an array (C-contiguous)
         * of some other type
         */
        struct _arr_descr *subarray;
        /*
         * The fields dictionary for this type
         * For statically defined descr this
         * is always Py_None
         */
        PyObject *fields;
        /*
         * An ordered tuple of field names or NULL
         * if no fields are defined
         */
        PyObject *names;
        /*
         * a table of functions specific for each
         * basic data descriptor
         */
        PyArray_ArrFuncs *f;
        /* Metadata about this dtype */
        PyObject *metadata;
        /*
         * Metadata specific to the C implementation
         * of the particular dtype. This was added
         * for NumPy 1.7.0.
         */
        NpyAuxData *c_metadata;
        /* Cached hash value (-1 if not yet computed).
         * This was added for NumPy 2.0.0.
         */
        npy_hash_t hash;
```

FWIW The docs you link to don't document metadata / c_metadata either.

**Comment by tacaswell** [comment]
I beleive this is closed by #13234.

**Comment by WarrenWeckesser** [comment]
Thanks @tacaswell.
